implementation/fault-tolerance/src/main/java/io/smallrye/faulttolerance/FaultToleranceInterceptor.java

```diff
@@ -158,6 +164,10 @@ public Object interceptCommand(InvocationContext invocationContext) throws Excep
                 CompletableFuture<T> result = new CompletableFuture<>();
                 asyncExecutor.submit(() -> {
                     try {
+                        // the requestContextController.activate/deactivate pair here is the minimum
+                        // to pass TCK; for anything serious, Context Propagation is required!
+                        // TODO does this code work together with Context Propagation?
+                        requestContextController.activate();
```

Comment:
I'd be mostly interested in your thoughts about this. I'm pretty sure the `Future`  case is fine, but here, I'm not sure.